storage/client.go

```diff
@@ -0,0 +1,39 @@
+package storage
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/nedpals/supabase-go"
+)
+
+const (
+	// signedURLExpiration is the expiration time of the signed URL in seconds
+	signedURLExpiration = 3600
+)
+
+type Client struct {
+	client *supabase.Client
+	bucket string
+}
+
+func NewClient(client *supabase.Client, bucket string) *Client {
+	return &Client{
+		client: client,
+		bucket: bucket,
+	}
+}
+
+func (c *Client) GenerateFileURL(filePath string) (url.URL, error) {
+	resp, err := c.client.Storage.From(c.bucket).CreateSignedURLForDownload(filePath, signedURLExpiration)
```

Comment:
Calling the Supabase Storage API and generating a new signed URL each time a release  is read from the database could result in heavy traffic to the Storage API (and increased response time). 

I think It would make sense to generate signed URLs with a longer expiration time (several hours, at least?) and cache them (using some in-memory package, e.g. [freecache](https://github.com/coocood/freecache), I think using Redis or something similar would be overkill, It's acceptable if the cache is lost when the application stops).

Another optimization could be generating signed URLs for all the files needed for releases in a single request. Supabase Storage offers an endpoint that supports this approach.

On the other hand, we're still dealing with a relatively small number of files in the case of ReleaseManager, so I wouldn't complicate it too much. Love to know your opinion.